warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IteratorBuildingVisitor.java

```diff
@@ -1327,19 +1332,11 @@ public FunctionFilter(List<ASTFunctionNode> nodes) {
         
         @Override
         public boolean keep(Key k) {
-            // fieldname is after fi\0
-            String fieldName = k.getColumnFamily().toString().substring(3);
             
-            // fieldvalue is first portion of cq
-            String fieldValue = k.getColumnQualifier().toString();
-            // pull off datatype and uid
-            int index = fieldValue.lastIndexOf('\0');
-            index = fieldValue.lastIndexOf('\0', index - 1);
-            fieldValue = fieldValue.substring(0, index);
+            parser.parse(k);
             
-            // create a jexl context with this valud
-            JexlContext context = new DatawaveJexlContext();
-            context.set(fieldName, new ValueTuple(fieldName, fieldValue, fieldValue, null));
+            context.clear();
+            context.set(parser.getField(), new ValueTuple(parser.getField(), parser.getValue(), parser.getValue(), null));
```

Comment:
It could be marginally faster to assign the results of `getField` and `getValue` to `final String` as opposed to calling these methods multiple times (e.g., skip the function lookup and the branch checks in the functions themselves)

Reply:
+1

Reply:
Done